File: src/entities/ProductionCompany.ts
import {
  Column,
  Entity,
  Index,
  OneToMany,
  PrimaryGeneratedColumn,
} from "typeorm";
import { MovieProductionCompany } from "./MovieProductionCompany";
import { SeriesProductionCompany } from "./SeriesProductionCompany";

@Index("pk_production_company", ["productionCompanyId"], { unique: true })
@Entity("production_company", { schema: "dionysus" })
export class ProductionCompany {
  @PrimaryGeneratedColumn({ type: "bigint", name: "production_company_id" })
  productionCompanyId: string;

  @Column("character varying", { name: "name", length: 100 })
  name: string;

  @Column("character varying", { name: "image", nullable: true, length: 480 })
  image: string | null;

  @OneToMany(
    () => MovieProductionCompany,
    (movieProductionCompany) => movieProductionCompany.productionCompany
  )
  movieProductionCompanies: MovieProductionCompany[];

  @OneToMany(
    () => SeriesProductionCompany,
    (seriesProductionCompany) => seriesProductionCompany.productionCompany
  )
  seriesProductionCompanies: SeriesProductionCompany[];
}

File: scrapper/scrapper.api/Entities/ProductionCompany.cs
using System;
using System.Collections.Generic;

namespace scrapper.api.Entities
{
    public partial class ProductionCompany
    {
        public ProductionCompany()
        {
            MovieProductionCompanies = new HashSet<MovieProductionCompany>();
            SeriesProductionCompanies = new HashSet<SeriesProductionCompany>();
        }

        public int ProductionCompanyId { get; set; }
        public string Name { get; set; } = null!;
        public string? Image { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }

        public virtual ICollection<MovieProductionCompany> MovieProductionCompanies { get; set; }
        public virtual ICollection<SeriesProductionCompany> SeriesProductionCompanies { get; set; }
    }
}
